src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -119,6 +126,8 @@ public JsonSchemaTransformer(MarkdownEngine markdownEngine, LinkResolver linkRes
                     LazyThreadSafetyMode.PublicationOnly);
             }
 
+            xref.Monikers = _monikerProvider.GetFileLevelMonikers(file.FilePath).monikers;
```

Comment:
consider make `monikers` a required constructor parameter of `InternalXrefSpec` to help avoid this kind of bug.